crates/astria-account-monitor/src/config.rs

```diff
@@ -70,6 +89,46 @@ impl Display for Asset {
     }
 }
 
+#[derive(Debug, Clone, Serialize)]
+pub struct SequencerAssetsToMonitor(Vec<Asset>);
+impl<'de> Deserialize<'de> for SequencerAssetsToMonitor {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let value = String::deserialize(deserializer)?;
+
+        if value.is_empty() {
+            return Err(serde::de::Error::custom("empty asset list"));
+        }
+
+        let assets: Result<Vec<Asset>, _> = value
```

Comment:
duplicate type information: no need to have `Result<Vec<Asset>, _>` both in the variable declaration and as type parameters in the `collect::<_, _>()` call below.